src/tdyutils.c

```diff
@@ -484,6 +484,31 @@ PetscInt TDyReturnIndexInList(PetscInt *list, PetscInt nlist, PetscInt value) {
 }
 
 /* -------------------------------------------------------------------------- */
+/// Outputs a sequential PETSc Vec as binary
+///
+/// @param [in] vec A sequential PETSc Vec
+/// @param [in] filename Name of the file
+/// @returns 0 on success, or a non-zero error code on failure
+PetscErrorCode TDySavePetscVecSeqAsBinary(Vec vec, const char filename[]) {
```

Comment:
I was going to say that I didn't think it was necessary to have a separate function for outputing a sequential  vector this way, but I can't find any way to get the communicator for a Vec object from the PETSc interface. So I guess we do need it!

Reply:
https://petsc.org/release/docs/manualpages/Sys/PetscObjectGetComm.html

This is also powerful, enabling `-some_prefix_vec_view vtk:that-vec.vtu` and lots of other kinds of viewing.
https://petsc.org/release/docs/manualpages/Sys/PetscObjectViewFromOptions.html

Reply:
Of course. Thanks, Jed. I had not thought to look "up the chain." I've been doing too much C++ lately.

@bishtgautam , in that case, we don't need an additional function for writing out vectors with binary viewers. We can just use the `PetscObjectGetComm` function to retrieve the MPI communicator in the existing function, and use that to open the viewer. It should be `PETSC_COMM_SELF` in the case of a serial vector.

And it's worth considering whether to just rely on `PetscObjectViewFromOptions` for viewing. It seems to offer a lot of flexibility.

Reply:
Note that `VecView(X, NULL)` uses stdout on a matching communicator. Useful for debugging, or `call VecView(X, 0)` in gdb.

Reply:
I advocate the `ViewFromOptions` strategy since it is so extensible

Reply:
@knepley @jedbrown , I'm trying to understand `VecViewFromOptions` in PETSc `src/snes/tutorials/ex12.c`

- `ierr = VecViewFromOptions(u, NULL, "-guess_vec_view");CHKERRQ(ierr);` is at [ex12.c#L862](https://gitlab.com/petsc/petsc/-/blob/main/src/snes/tutorials/ex12.c#L862)
- Then looking at a test on [ex12.c#L1636](https://gitlab.com/petsc/petsc/-/blob/main/src/snes/tutorials/ex12.c#L1636) that includes `-guess_vec_view glvis:`

By specifying `-guess_vec_view glvis:`, is the Vec being written out in a `glvis` format? Is the output displayed on the screen because there isn't a string specified after `:` for the output to be saved to a file?

Reply:
Yeah, I think you'd need a glvis server running (to receive and render this data). The option is powerful in providing many different ways to view.